### PR TITLE
Add logging variant for netcdf-c

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds the logging option for netcdf-c. Tested builds to ensure the configure and cmake logging settings work, but didn't test the logging itself.

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by skosukhin** [comment]
Can this wait until next week?

**Comment by alalazo** [comment]
Sure @skosukhin Feel free to ping me for a merge when ready.

**Comment by AlexanderRichert-NOAA** [comment]
@skosukhin have you had a chance to look at this yet?

**Review by edwardhartnett — APPROVED** [review]

File: var/spack/repos/builtin/packages/netcdf-c/package.py [hunk 1]
```diff
@@ -530,6 +532,8 @@ def configure_args(self):
             # Suppress the redundant check for m4:
             config_args.append("ac_cv_prog_NC_M4=false")
 
+        config_args.extend(self.enable_or_disable("logging"))
```
Comment (skosukhin) [inline var/spack/repos/builtin/packages/netcdf-c/package.py]: This is a suboptimal place in the code to put this. Please, move it:
```patch
--- a/var/spack/repos/builtin/packages/netcdf-c/package.py
+++ b/var/spack/repos/builtin/packages/netcdf-c/package.py
@@ -434,6 +434,8 @@ def configure_args(self):
 
         config_args += self.enable_or_disable("fsync")
 
+        config_args += self.enable_or_disable("logging")
+
         if any(self.spec.satisfies(s) for s in ["+mpi", "+parallel-netcdf", "^hdf5+mpi~shared"]):
             config_args.append("CC={0}".format(self.spec["mpi"].mpicc))
 
@@ -532,8 +534,6 @@ def configure_args(self):
             # Suppress the redundant check for m4:
             config_args.append("ac_cv_prog_NC_M4=false")
 
-        config_args.extend(self.enable_or_disable("logging"))
-
         lib_search_dirs.extend(d for libs in extra_libs for d in libs.directories)
         # Remove duplicates and system prefixes:
         lib_search_dirs = filter_system_paths(dedupe(lib_search_dirs))
```

**Review by skosukhin — CHANGES_REQUESTED** [review]
I remember there was something inconsistent about this option but I can't recall what it was and I can't find it now. Therefore, LGTM as soon as the suggested decorative change is made.

File: var/spack/repos/builtin/packages/netcdf-c/package.py (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/netcdf-c/package.py]: done, thanks

**Review by alalazo — DISMISSED** [review]
@skosukhin

File: var/spack/repos/builtin/packages/netcdf-c/package.py [hunk 2]
```diff
@@ -432,6 +434,8 @@ def configure_args(self):
 
         config_args += self.enable_or_disable("fsync")
 
+        config_args.extend(self.enable_or_disable("logging"))
```
Comment (skosukhin) [inline var/spack/repos/builtin/packages/netcdf-c/package.py]: This is nitpicking but shall we keep the code homogeneous as in my previous request?

```suggestion
        config_args += self.enable_or_disable("logging")
```

@alalazo please, feel free to merge this as-is if you think that I'm asking for too much :)

**Review by alalazo — APPROVED** [review]

**Pull request merged**